tool/tsh/proxy.go

```diff
@@ -255,6 +258,91 @@ func mkLocalProxyCerts(certFile, keyFile string) ([]tls.Certificate, error) {
 	return []tls.Certificate{cert}, nil
 }
 
+func onProxyCommandAPP(cf *CLIConf) error {
+	tc, err := makeClient(cf, false)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	appCerts, err := loadAppCertificate(tc, cf.AppName)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	address, err := utils.ParseAddr(tc.WebProxyAddr)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	addr := "localhost:0"
+	if cf.LocalProxyPort != "" {
+		addr = fmt.Sprintf("127.0.0.1:%s", cf.LocalProxyPort)
+	}
+
+	listener, err := net.Listen("tcp", addr)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	lp, err := alpnproxy.NewLocalProxy(alpnproxy.LocalProxyConfig{
+		Listener:           listener,
+		RemoteProxyAddr:    tc.WebProxyAddr,
+		Protocol:           alpncommon.ProtocolHTTP,
+		InsecureSkipVerify: cf.InsecureSkipVerify,
+		ParentContext:      cf.Context,
+		SNI:                address.Host(),
+		AWSCredentials:     nil,
+		Certs:              []tls.Certificate{appCerts},
+	})
+	if err != nil {
+		if cerr := listener.Close(); cerr != nil {
+			return trace.NewAggregate(err, cerr)
+		}
+		return trace.Wrap(err)
+	}
+
+	defer lp.Close()
+	go func() {
+		if err = lp.Start(cf.Context); err != nil {
+			log.WithError(err).Errorf("Failed to start local proxy.")
+		}
+	}()
+
+	fmt.Printf("Proxying connections to %s on %v\n", cf.AppName, lp.GetAddr())
+
+	<-cf.Context.Done()
+
+	return nil
+}
+
+func loadAppCertificate(tc *client.TeleportClient, appName string) (tls.Certificate, error) {
+	key, err := tc.LocalAgent().GetKey(tc.SiteName, client.WithAppCerts{})
```

Comment:
Seems like kind of a bummer that `client.WithAppCerts` doesn't export `appName`. You could then only load the certs for the app you want instead of all apps. Not sure why that is but probably not something you want to change in this PR.